addons/sms/models/mail_thread.py

```diff
@@ -42,34 +42,15 @@ def _compute_message_has_sms_error(self):
     def _search_message_has_sms_error(self, operator, operand):
         return ['&', ('message_ids.has_sms_error', operator, operand), ('message_ids.author_id', '=', self.env.user.partner_id.id)]
 
-    def _sms_get_partner_fields(self):
-        """ This method returns the fields to use to find the contact to link
-        whensending an SMS. Having partner is not necessary, having only phone
-        number fields is possible. However it gives more flexibility to
-        notifications management when having partners. """
-        fields = []
-        if hasattr(self, 'partner_id'):
-            fields.append('partner_id')
-        if hasattr(self, 'partner_ids'):
-            fields.append('partner_ids')
-        return fields
-
     def _sms_get_default_partners(self):
         """ This method will likely need to be overridden by inherited models.
                :returns partners: recordset of res.partner
         """
         partners = self.env['res.partner']
-        for fname in self._sms_get_partner_fields():
+        for fname in self._mail_get_partner_fields():
             partners = partners.union(*self.mapped(fname))  # ensure ordering
         return partners
 
-    def _sms_get_number_fields(self):
```

Comment:
Note that "_phone_get_number_fields" returns an empty list by default, and this one returns mobile if found, so it alters the behavior. Can that lead to issues? Shouldn't we keep the fallback?

EDIT: welp, looks like it's done in the next commit ^^